docs/commands.rst

```diff
@@ -0,0 +1,213 @@
+Advanced: PyOTA Commands
+========================
+
+.. note::
+    **This page contains information about how PyOTA works under the hood.**
+
+    It is absolutely not necessary to be familiar with the content described
+    below if you just want to use the library.
+
+    However, if you are a curious mind or happen to do development on the
+    library, the following information might be useful.
+
+PyOTA provides the API interface (:ref:`Core API Methods` and
+:ref:`Extended API Methods`) for users of the library. These handle
+constructing and sending HTTP requests to the specified node through adapters,
+furthermore creating, transforming and translating between PyOTA specific types
+and (JSON-encoded) raw data. They also filter outgoing requests and incoming
+responses to ensure that only appropriate data is communicated with the node.
+
+The high level API interface methods (:ref:`Core API Methods` and
+:ref:`Extended API Methods`) internally call PyOTA commands to get the job done.
+
+Most PyOTA commands are sub-classed from :py:class:`FilterCommand` class, which
+is in turn sub-classed from :py:class:`BaseCommand` class. The reason for the
+2-level inheritance is simple: separating functionality. As the name implies,
+:py:class:`FilterCommand` adds filtering capabilities to
+:py:class:`BaseCommand`, that contains the logic of constructing the request
+and using its adapter to send it and receive a response.
+
+:py:class:`BaseCommand` is also decorated with a :py:class:`CommandMeta`
+metaclass, that helps to register each command automatically in the API classes'
+command registry at runtime.
```

Comment:
💭 We could probably deprecate and remove this functionality; I don't think anybody uses the command registries.

Short version is, I wanted to make it so that developers didn't have to add methods to the `Iota`/`StrictIota`/`MultisigIota`/etc. classes explicitly.  Instead, the registry would automatically load commands in the specified packages and attach them magically.

In exchange for being clever, this approach had a number of drawbacks:
* Very difficult to understand which commands a particular API class can use via static analysis.
* Difficult to document how users should invoke each command, what parameters to provide, etc.
* IDEs weren't couldn't support jumping from a command invocation to the corresponding class (e.g., `self.getNodeInfo()` \[via `__getattr__`\] vs. `core.GetNodeInfoCommand(self.adapter)()`).

In #30 I laid the groundwork for making command registries redundant.

Reply:
I removed command registries and modified command test cases. take a look on https://github.com/iotaledger/iota.py/pull/275/commits/dbb4de7198101dd9f99e78794f850f09c3b24302 .

Reply:
Awesome!  I added some comments to that commit.